### ISSUE TITLE
Windows msvc support

### ISSUE THREAD

**Issue by silverqx** [Chocobo1/setup-ccache-action#8]
Hi choco 👋,

this week I have set up the `ccache` for all my Windows msvc toolchains, it has **100% cache hits** and now I want to also set up the ccache for my Windows msvc related GitHub actions.

The following doesn't work, it quits with the `Warning: "windows_compile_environment=" is not supported. No operation...`:
```
    - name: ccache install 🥳
      run: |
        choco install ccache -y

    - name: ccache setup 🕺
      uses: Chocobo1/setup-ccache-action@v1
      with:
        prepend_symlinks_to_path: false
        install_ccache: false
        update_packager_index: false
        ccache_options: |
          compression = false
          max_size = 1G
```

I don't ask for full msvc support, even the `install_ccache` may not work.

Would be good to normally continue if the `windows_compile_environment` is empty or `windows_compile_environment=msvc`.

I have tried following inside the `Windows-2019` GitHub action's image, I'm posting here notes to make it easier for you:
- The `ccache --set-config xx=yy` works out of the box if the `ccache` is already installed.
- it creates `ccache.conf` file at `$env:APPDATA/ccache/ccache.conf` (~/AppData/Roaming/ccache)
- choco installs `ccache.exe` at `C:\ProgramData\chocolatey\bin\ccache.exe` and it's on the system path

I think it would be enough to make it work inside the `Windows-2019/2022` images.

[Here](https://github.com/silverqx/TinyORM/blob/main/cmake/CommonModules/TinyHelpers.cmake#L314-L401) is my cmake code that fixes/allows 100% ccache hits, it simply disables precompiled headers and replaces `/Zi` compiler option with the `/Z7` in `CMAKE_<C|CXX>_FLAGS_<CONFIG>` flags.
The Cmake defaults to `/Zi` so it has 100% ccache misses because ccache doesn't support `/Zi` but it works great with the `/Z7` compiler option. And also ccache doesn't support precompiled headers on Windows msvc.

**Comment by silverqx** [comment]
Currently, I have workarounded it using (for inspiration 👀):
```cmake
- name: Ccache restore cache 🕺
  id: cache-ccache-msvc2022-qt63
  uses: actions/cache@v3
  env:
    cache-name: msvc2022-qt63
  with:
    path: ~/AppData/Roaming/ccache
    key: ${{ runner.os }}-ccache-${{ env.cache-name }}-${{ github.run_id }}
    restore-keys: |
      ${{ runner.os }}-ccache-${{ env.cache-name }}-

- name: Ccache setup 🥳
  run: |
    echo "::group::Install ccache"
    choco install ccache -y
    echo "::endgroup::"

    echo "::group::Prepare ccache config"
    ccache --set-config max_size=600M
    ccache --set-config sloppiness=pch_defines,time_macros
    ccache --show-config
    echo "::endgroup::"

    echo "::group::Clear ccache statistics"
    ccache --zero-stats
    echo "::endgroup::"

- name: TinyORM cmake configure (msvc-cmake-debug)
  run: >-
    cmake
    -S .
    -B ../TinyORM-builds-cmake/build-msvc-cmake-debug
    -G Ninja
    -D CMAKE_CXX_COMPILER_LAUNCHER:FILEPATH="C:/ProgramData/chocolatey/bin/ccache.exe"
    -D CMAKE_TOOLCHAIN_FILE:FILEPATH="${env:VCPKG_INSTALLATION_ROOT}/scripts/buildsystems/vcpkg.cmake"
    -D CMAKE_DISABLE_PRECOMPILE_HEADERS:BOOL=ON
    -D CMAKE_BUILD_TYPE:BOOL=Debug
    ...

- name: TinyORM cmake build ✨ (msvc-cmake-debug)
  run: |
    cmake --build ../TinyORM-builds-cmake/build-msvc-cmake-debug --target all --parallel 2

- name: Ccache statistics
  run: |
    ccache.exe --show-stats -vv
```

**Comment by Chocobo1** [comment]
Hi, sorry for the late reply. I've just released v1.2.11 which should add some support to MSVC.
Also thanks for your instructions, it helps immensely when coding this feature.
